src/Orleans/IDs/Interner.cs

```diff
@@ -245,14 +252,17 @@ private void InternCacheCleanupTimerCallback(object state)
                     }
                 }
             }
-            long numRemoved = numEntries - internCache.Count;
+
+            if (!removalResultsLoggingNeeded) return;
+
+            var numRemoved = numEntries - internCache.Count;
             if (numRemoved > 0)
             {
                 if (logger.IsVerbose) logger.Verbose(ErrorCode.Runtime_Error_100296, "Removed {0} / {1} unused {2} entries in {3}", numRemoved, numEntries, internCacheName, clock.Elapsed);
             }
             else
             {
-                if (logger.IsVerbose2) logger.Verbose2(ErrorCode.Runtime_Error_100296, "Removed {0} / {1} unused {2} entries in {3}", numRemoved, numEntries, internCacheName, clock.Elapsed);
+                logger.Verbose2(ErrorCode.Runtime_Error_100296, "Removed {0} / {1} unused {2} entries in {3}", numRemoved, numEntries, internCacheName, clock.Elapsed);
```

Comment:
I know it redundant, but lets keep`if (logger.IsVerbose2)` here, just in case to be robust against future changes.